lib/mongoose.js

```diff
@@ -292,6 +294,16 @@ Mongoose.prototype.set = function(key, value) {
       } else if (!optionValue && _mongoose.transactionAsyncLocalStorage) {
         delete _mongoose.transactionAsyncLocalStorage;
       }
+    } else if (optionKey === 'createInitialConnection') {
+      if (optionValue && !_mongoose.connection) {
+        const conn = this.createConnection(); // default connection
+        conn[defaultConnectionSymbol] = true;
+        conn.models = this.models;
```

Comment:
i dont know if this will be modified often, but maybe consider putting this code in a function to de-duplicate default connection creation

Reply:
Fixed in c04a75d